src/coreclr/tools/Common/TypeSystem/Common/TargetDetails.cs

```diff
@@ -23,7 +23,8 @@ public enum TargetOS
         FreeBSD,
         NetBSD,
         SunOS,
-        WebAssembly
+        WebAssembly,
+        Android
```

Comment:
Why does the AOT compiler need to distinguish between Linux and Android? We've been shipping for Android for a while now with native AOT and it's just Linux there.

Reply:
There are two possible scenarios:

1. We'll just treat it as Linux everywhere in the compiler anyway. If that's the case, we should just normalize to linux before compiler is invoked.
2. The compiler needs to know this is Android. Then this change makes sense.

We had a similar discussion around iOSLike vs macOS in https://github.com/dotnet/runtime/issues/87610#issuecomment-1639993541. That one even wanted to add extra enums to RyuJIT to distinguish. We ended up not doing it because there was nothing different.

The AOT compiler (but not RyuJIT) needs to distinguish between macOS and iOSLike because Apple really wants to see that in the object files. I'm not aware of a need for this in the Android object files.

Reply:
I'd prefer to hedge towards needing a change in the aot compiler and then back it out if we don't. Once we are able to run something (perhaps in this PR), we might have a better idea.